src/utils/testing.tsx

```diff
@@ -247,8 +247,25 @@ export function checkStoriesForAxeViolations(name: string, storyDir?: string) {
     if (typeof Story !== 'function') return
 
     const {storyName, name: StoryFunctionName} = Story as StoryType
+
+    beforeEach(() => {
+      // IntersectionObserver isn't available in test environment
+      const mockIntersectionObserver = jest.fn()
+      mockIntersectionObserver.mockReturnValue({
+        observe: () => null,
+        unobserve: () => null,
+        disconnect: () => null,
+      })
+      window.IntersectionObserver = mockIntersectionObserver
+    })
+
     it(`story ${storyName || StoryFunctionName} should have no axe violations`, async () => {
-      const {container} = HTMLRender(<Story />)
+      const {container} = HTMLRender(
```

Comment:
[checkStoriesForAxeViolation](https://github.com/primer/react/blob/main/src/PageHeader/PageHeader.test.tsx#L182) was failing on `PageHeader` after adding the new `WithPageLayout` story, because it wasn't able to access the theme object in `PageLayout` on lines like this 
```
[`@media screen and (min-width: ${theme.breakpoints[1]})`]
```

So I wrapped the `<Story />` with the `<ThemeProvider></ThemeProvider>` here and also needed to mock the IntersectionObserver. Tests are succeeding now. Let me know if there is a better way to do this 🙌🏼

Reply:
Not that I know of. This seems like a reasonable solution to me.